### PR TITLE
remove json and url encoded form support from -http

### PR DESCRIPTION
**Pull request by robjtede**
## PR Type
Dep Reduction

## PR Checklist
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview
Further reduce the number of dependencies on -http, making it simpler.

Removes some `Into<Error>` and `ResponseError` impls for each lib's error types. I think this makes sense anyway but need to look at the error system before v4 anyhow, it's possible the response error trait can be moved to -web, too, which would make it possible to re-impl for these types.

### CODE REVIEW

File: awc/src/sender.rs [hunk 1]
```diff
@@ -209,7 +209,8 @@ impl RequestSender {
     ) -> SendClientRequest {
         let body = match serde_json::to_string(value) {
             Ok(body) => body,
-            Err(e) => return Error::from(e).into(),
+            // TODO: own error type
+            Err(e) => return Error::from(io::Error::new(io::ErrorKind::Other, e)).into(),
```
Comment (robjtede) [inline awc/src/sender.rs:213]: Thoughts on what do do here @fakeshadow, maybe a wrapped error that can also hold serialization errors as well as client errors from -http. Might leave it for a separate PR if you want to handle it.

File: actix-http/src/response.rs [hunk 2]
```diff
@@ -533,32 +454,10 @@ impl ResponseBuilder {
         self.body(Body::from_message(BodyStream::new(stream)))
     }
 
-    /// Set a json body and generate `Response`
-    ///
-    /// `ResponseBuilder` can not be used after this call.
-    pub fn json(&mut self, value: impl Serialize) -> Response {
```
Comment (popzxc) [inline actix-http/src/response.rs:539]: Won't it make sense to leave this method under an optional feature `json`? If the project aims to get rid of it completely, this feature can be declared deprecated from the start.

There may be code that relies on this method, thus removing it completely will make migration harder; I think it will be a bit more friendly to users to provide some way to start migrating without introducing way too many changes.

**Review by popzxc — APPROVED** [review]
In general LGTM.

File: actix-http/src/response.rs (hunk 2 above)
Comment (robjtede) [reply actix-http/src/response.rs:539]: In terms of migration, it should be a non-issue for most: `actix_web::HttpResponse` was a re-export from `actix_http::Response`. Now actix-web has it's own HttpResponse object that keeps the json method.

Comment (popzxc) [reply actix-http/src/response.rs:539]: Ah, I see, it totally makes sense 👍

**Comment by tjkirch** [comment]
Howdy!  Bit of feedback on this, particularly the error changes, since it wasn't clear to me how to operate without those helpers.  Up through beta.5, I was able to do things like `HttpResponse::BadRequest().body(err.to_string()).into()` in an `error_handler`, but now there would be more manual conversion required, since the `Into` is gone.  I hope that if the error changes stay in place for v4 final, there can be some wording in the docs/changelog about how to adapt existing code.  (I had to update from v3 to v4 beta to be able to update tokio.)
